packages/react-router/lib/router/history.ts

```diff
@@ -510,7 +510,7 @@ export function warning(cond: any, message: string) {
 }
 
 function createKey() {
-  return Math.random().toString(36).substr(2, 8);
+  return Math.random().toString(36).substring(2, 8);
```

Comment:
This will produce a different result. The 2nd arg in `substr` is a length, whereas in `substring` it is the ending index for the substring. This will produce a 6 character string versus the original 8 character string.

Reply:
Ah, I see. I will make required changes and close this PR.